### PR TITLE
doc: break down and refine docs

### PR DESCRIPTION
**Pull request by yolocs**
- Added Go examples but need a little help on Java examples (will follow up).
- Pending less user facing docs (testing and setup)

### CODE REVIEW

File: docs/clients.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Clients
+
+**Lumberjack is not an official Google product.**
+
+Lumberjack client consists of the following processors and they run in order:
+
+1.  Validator(s): To validate audit log requests
+2.  Mutator(s): To mutate audit log request, e.g. applying common labels
```
Comment (myurtoglu) [inline docs/clients.md:8]: Maybe adding a "the" here to make it "To mutate the audit log request," is a good idea.

File: docs/clients.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  regex:
+    principal_include: "@example1\\.com$|@example2\\.com$"
+backend:
+  cloudlogging:
+    default_project: true
+labels:
+  common_label_1: foobar
+```
+
+### Exend
```
Comment (myurtoglu) [inline docs/clients.md:69]: Extend?

File: docs/config.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+Please refer to the [gRPC guide](./grpc.md).
+
+## (WIP) Justification
+
+TODO
+
+## Supported env vars
+
+| ENV VAR name                                      | Description              |
```
Comment (myurtoglu) [inline docs/config.md:90]: This table is looking a bit weird in github code review's preview, hopefully it will render ok?

**Review by myurtoglu — APPROVED** [review]

File: docs/config.md (hunk 3 above)
Comment (yolocs) [reply docs/config.md]: It seems github markdown doesn't recognize this table syntax. Fixed now.

**Pull request merged**